ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+/**
+ * @module DashboardQuickActionsCard
+ * DashboardQuickActionsCard component allows users to see a list of secrets engines filtered by
+ * kv, pki and database and perform certain actions based on the type of secret engine selected
+ *
+ * @example
+ * ```js
+ * <DashboardQuickActionsCard @secretsEngines={{@model.secretsEngines}} />
+ * ```
+ * @param {array}
+ */
+
+const getActionsByEngineType = (type) => {
+  switch (type) {
+    case 'kv':
+      return [
+        {
+          actionTitle: 'Find KV secrets',
+          actionType: 'find-kv',
+        },
+      ];
+    case 'database':
+      return [
+        {
+          actionTitle: 'Generate credentials for database',
```

Comment:
do we need both `actionTitle` and `actionType` here? wondering if we could just return `actionTitle` and use that to compute the getter below so the switch value would just be the title